flowzone.yml

```diff
@@ -1213,38 +1213,39 @@ jobs:
             "$(npm root -g)"/versionist/scripts/generate-changelog.sh .
           fi
 
-      # run balena-versionist
-      # print the error if anything fails
-      # fail the workflow if the error is anything other than "No such file or directory"
-      - name: Run versionist
+      # https://github.com/actions/github-script
+      - name: Run balena-versionist
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7
         if: inputs.disable_versioning != true
         id: versionist
         env:
+          # Use to authenticate with GitHub for private nested changelogs
           GITHUB_TOKEN: ${{ steps.gh_app_token.outputs.token || secrets.FLOWZONE_TOKEN }}
-        run: |
-          out="$(balena-versionist 2>&1)"
-          error="$(awk '/Error:/{getline; print}' <<< "${out}")"
-
-          case ${error} in
-            "") # no error
-              ;;
-            'No such file or directory'*'/package.json')
-              echo "::error file=.versionbot/CHANGELOG.yml,line=1::Versionist expects a package.json if repo.yml does not provide a 'type' for the project"
-              ;;
-            *)
-              echo "::error::${error}"
-              exit 1
-              ;;
-          esac
-
-          git status --porcelain
+        with:
+          result-encoding: json
+          script: |
+            const path = require('path');
+            const { execSync } = require('child_process');
 
-          versions=()
-          [ -f .versionbot/CHANGELOG.yml ] && read -r -a versions < <(yq e '.[0].version' .versionbot/CHANGELOG.yml)
-          semver="${versions[0]}"
+            // Get global node_modules path
+            const globalNodeModules = execSync('npm root -g').toString().trim();
+            const { runBalenaVersionist } = require(path.join(globalNodeModules, 'balena-versionist'));
```

Comment:
this seems very convoluted to search for the global module to import it. Isn't there a way to import external modules from github script?

Reply:
Yeah, it seemed weird to me too. Let me see if I can find the recommended approach.

Reply:
I switched to setting NODE_ENV once in the job environment and that seems cleaner.